legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-snowflake/src/main/java/org/finos/legend/engine/persistence/components/relational/snowflake/sql/visitor/DigestUdfVisitor.java

```diff
@@ -36,14 +39,29 @@ public VisitorResult visit(PhysicalPlanNode prev, DigestUdf current, VisitorCont
     {
         Udf udf = new Udf(context.quoteIdentifier(), current.udfName());
         prev.push(udf);
-        List<Value> values = new ArrayList<>();
+        List<Value> columnNameList = new ArrayList<>();
+        List<Value> columnValueList = new ArrayList<>();
         for (int i = 0; i < current.values().size(); i++)
         {
-            values.add(StringValue.of(current.fieldNames().get(i)));
-            values.add(current.values().get(i));
+            columnNameList.add(StringValue.of(current.fieldNames().get(i)));
+            columnValueList.add(getTypeConversionUdf(current.values().get(i), current.fieldTypes().get(i), current.typeConversionUdfNames()));
         }
 
-        FunctionImpl function = FunctionImpl.builder().functionName(FunctionName.OBJECT_CONSTRUCT).addAllValue(values).build();
-        return new VisitorResult(udf, Arrays.asList(function));
+        Function toArrayColumnNames = FunctionImpl.builder().functionName(FunctionName.ARRAY_CONSTRUCT).addAllValue(columnNameList).build();
+        Function toArrayColumnValues = FunctionImpl.builder().functionName(FunctionName.ARRAY_CONSTRUCT).addAllValue(columnValueList).build();
+
+        return new VisitorResult(udf, Arrays.asList(toArrayColumnNames, toArrayColumnValues));
+    }
+
+    private Value getTypeConversionUdf(Value value, DataType dataType, Map<DataType, String> typeConversionUdfNames)
```

Comment:
Can we move the if else check in getTypeConversionUdf() if possible ??
then we can possibly have the same logic, ignore otherwise if it's not feasible

Reply:
Sure, changed! Added a DigestUdfVisitor in ANSI module so that they can inherit from it, but I left it unregistered in ANSI sink so that it cannot be invoked from ANSI sink itself. Let me know if the approach is fine